src/components/QueryEditor.tsx

```diff
@@ -143,6 +146,7 @@ const QueryClause = ({
         <AutocompleteInput
           value={con.relation}
           setValue={setConditionRelation}
+          onBlur={(e) => setConditionRelation(e, false)}
```

Comment:
Id be ok with calling `setConditionRelation` directly instead of the other usage, then not needing the timeout param

Reply:
Do you mean dropping `setValue` and the timeout altogether?

Reply:
ooh hm. okay let's see how keeping both fares for now